dropwizard-index-page/src/main/java/com/palantir/indexpage/ClasspathIndexPage.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ */
+
+package com.palantir.indexpage;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.github.mustachejava.DefaultMustacheFactory;
+import com.github.mustachejava.Mustache;
+import com.github.mustachejava.MustacheFactory;
+import com.google.common.base.Charsets;
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableMap;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.StringWriter;
+import java.net.URL;
+import java.util.Map;
+
+/**
+ * Used to represent the index page loaded from the classpath.
+ */
+public final class ClasspathIndexPage implements TemplatedAsset {
+
+    private final Mustache mustacheCompiler;
+
+    private Map<String, String> context;
+    private StringWriter writer;
+
+    public ClasspathIndexPage(URL resource) {
+        checkNotNull(resource);
+
+        MustacheFactory factory = new DefaultMustacheFactory();
+        BufferedReader reader;
+        try {
+            reader = new BufferedReader(new InputStreamReader(resource.openStream(), Charsets.UTF_8));
+        } catch (IOException e) {
+            throw new RuntimeException("Index page cannot be found at " + resource.getPath(), e);
+        }
+
+        mustacheCompiler = factory.compile(reader, IndexPageBundle.INDEX_PAGE_NAME);
+        writer = new StringWriter();
+    }
+
+    @Override
+    public Optional<String> getContent(Map<String, String> inputContext) {
```

Comment:
nit: think about renaming `inputContext` mostly because of the clash with the `contextPath` concept.